differs/pipDiff.go

```diff
@@ -37,27 +39,50 @@ func getPythonVersion(pathToLayer string) ([]string, error) {
 	return matches, nil
 }
 
-func (d PipDiffer) getPackages(path string) (map[string]map[string]utils.PackageInfo, error) {
-	packages := make(map[string]map[string]utils.PackageInfo)
+func getPythonPaths(vars []string) []string {
+	paths := []string{}
+	for _, envVar := range vars {
+		pythonPathPattern := regexp.MustCompile("^PYTHONPATH=(.*)")
+		match := pythonPathPattern.FindStringSubmatch(envVar)
+		if len(match) != 0 {
+			pythonPath := match[1]
+			paths = strings.Split(pythonPath, ":")
+			break
+		}
+	}
+	return paths
+}
 
+func (d PipDiffer) getPackages(image utils.Image) (map[string]map[string]utils.PackageInfo, error) {
+	path := image.FSPath
+	packages := make(map[string]map[string]utils.PackageInfo)
+	pythonPaths := []string{}
+	if !reflect.DeepEqual(utils.ConfigSchema{}, image.Config) {
+		paths := getPythonPaths(image.Config.Config.Env)
+		for _, p := range paths {
+			pythonPaths = append(pythonPaths, p)
+		}
+	}
 	pythonVersions, err := getPythonVersion(path)
 	if err != nil {
-		// layer doesn't have a Python version installed
+		// Image doesn't have Python installed
 		return packages, nil
 	}
 
-	for _, pyVersion := range pythonVersions {
-		packagesPath := filepath.Join(path, "usr/local/lib", pyVersion, "site-packages")
-		contents, err := ioutil.ReadDir(packagesPath)
+	for _, pythonVersion := range pythonVersions {
+		packagesPath := filepath.Join(path, "usr/local/lib", pythonVersion, "site-packages")
+		pythonPaths = append(pythonPaths, packagesPath)
+	}
+
+	for _, pythonPath := range pythonPaths {
+		contents, err := ioutil.ReadDir(pythonPath)
 		if err != nil {
-			// layer's Python folder doesn't have a site-packages folder
```

Comment:
Oh hmm, you're right, if I'm preemptively building those "site-packages" paths without knowing for sure if they exist, I should just skip over them if they don't exist... I mostly deleted it because it referenced layers and presumably, there should be no layers once we address #11